### PR TITLE
update cli support for dataconnect toolkit and webhooks

### PR DESCRIPTION
**Pull request by hlshen**

### CODE REVIEW

File: src/emulator/dataconnectToolkitController.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { EmulatorInstance, EmulatorInfo } from "./types";
+import { FirebaseError } from "../error";
+import * as portUtils from "./portUtils";
+import * as express from "express";
```
Comment (joehan) [inline src/emulator/dataconnectToolkitController.ts:4]: Remove

File: src/emulator/hub.ts [hunk 2]
```diff
@@ -50,7 +51,8 @@ export class EmulatorHub extends ExpressBasedEmulator {
     const data = fs.readFileSync(locatorPath, "utf8").toString();
     const locator = JSON.parse(data) as Locator;
 
-    if (locator.version !== this.CLI_VERSION) {
+    // TODO: In case the locator file format is change, handle issues with format incompatability
```
Comment (joehan) [inline src/emulator/hub.ts:54]:
```suggestion
    // TODO: In case the locator file format is changed, handle issues with format incompatibility
```

**Review by joehan — COMMENTED** [review]
Went through this together in person, and its very close to LGTM. I'd love to get some of the low hanging fruit TODOs in tho

File: src/emulator/commandUtils.ts [hunk 3]
```diff
@@ -464,9 +465,11 @@ export async function emulatorExec(script: string, options: any): Promise<void>
   try {
     const showUI = !!options.ui;
     ({ deprecationNotices } = await controller.startAll(options, showUI, true));
+    sendVSCodeMessage({ message: VSCODE_MESSAGE.EMULATORS_STARTED });
     exitCode = await runScript(script, extraEnv);
     await controller.onExit(options);
   } finally {
+    sendVSCodeMessage({ message: VSCODE_MESSAGE.EMULATORS_START_ERRORED });
```
Comment (rrousselGit) [inline src/emulator/commandUtils.ts:472]: Wouldn't having this in the `finally` block _always_ notify the emulator as errored?

Comment (joehan) [reply src/emulator/commandUtils.ts:472]: +1 - this should be in a new catch, and there should be a separate finally block for controller.cleanShutdown()

**Review by joehan — APPROVED** [review]
LGTM after the requested fixes

File: src/emulator/commandUtils.ts (hunk 3 above)
Comment (hlshen) [reply src/emulator/commandUtils.ts:472]: Oh I just didn't read that at all. good finally, I mean catch

**Review by joehan — APPROVED** [review]

**Pull request merged**